pkg/dcmtime/da_test.go

```diff
@@ -0,0 +1,83 @@
+package dcmtime
+
+import (
+	"github.com/stretchr/testify/assert"
+	"testing"
+	"time"
+)
+
+func TestDA_String(t *testing.T) {
+	testCases := []struct {
+		Time      time.Time
+		Precision PrecisionLevel
+		Expected  string
+	}{
+		{
+			Time: time.Date(
+				1010,
+				2,
+				3,
+				0,
+				0,
+				0,
+				0,
+				time.UTC,
+			),
+			Precision: Precision.Full,
+			Expected:  "10100203",
+		},
+		{
+			Time: time.Date(
+				1010,
+				2,
+				3,
+				0,
+				0,
+				0,
+				0,
+				time.UTC,
+			),
+			Precision: Precision.Day,
+			Expected:  "10100203",
+		},
+		{
+			Time: time.Date(
+				1010,
+				2,
+				3,
+				0,
+				0,
+				0,
+				0,
+				time.UTC,
+			),
+			Precision: Precision.Month,
+			Expected:  "101002",
+		},
+		{
+			Time: time.Date(
+				1010,
+				2,
+				3,
+				0,
+				0,
+				0,
+				0,
+				time.UTC,
+			),
+			Precision: Precision.Year,
+			Expected:  "1010",
+		},
+	}
+
+	for _, tc := range testCases {
+		name := tc.Expected + "_" + tc.Precision.String()
+		t.Run(name, func(t *testing.T) {
+			assert := assert.New(t)
+
+			da := NewDA(tc.Time, tc.Precision)
+
+			assert.Equal(da.DCM(), tc.Expected)
```

Comment:
instead of using assert here, can we use native testing checks? (Go team prefers to not use assertion libraries, and instead use native Go code as much as possible in tests, which is the convention elsewhere in the package as well. It's nice because it's one less dependency, and good 'ol go code without needing to worry about libraries. A little more boilerplate, but also lets you write better failures messages with appropriate info like what the inputs to the function of interest were. See https://golang.org/doc/faq#testing_framework for more thoughts on this). 

```suggestion
		t.Run(name, func(t *testing.T) {
			if da := NewDA(tc.Time, tc.Precision); da.DCM() != tc.Expected {
			    t.Errorf("NewDate(%v, %v): unexpected DCM output. got: %v, want: %v", tc.Time, tc.Precision, da.DCM(), tc.Expected)
			}
```